### PR TITLE
Update the GitLab Pages documentation

### PR DESCRIPTION
**Pull request by walpox**
Sanity check:

* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Rationale

1. What's stated in the section **Repository setup** seems to be outdated. As per the current documentation on GitLab, GitLab Pages does not seem to work the way described there.
2. In the GitLab CI template, the `only` keyword is no longer supported by GitLab. Its replacement is `rules`.
3. The GitLab CI template can be changed to allow using specific versions of Zola.
4. Rewritten some parts as per my personal writing style.
5. Some other changes were proposed by my Markdown linter, such as [removing the dollar signs from the fenced code blocks](https://github.com/DavidAnson/markdownlint/blob/main/doc/md014.md).
6. Removed comments that described _what_ a command does, instead of the _why_.

### CODE REVIEW

File: docs/content/documentation/deployment/gitlab-pages.md [hunk 1]
```diff
@@ -3,68 +3,91 @@ title = "GitLab Pages"
...
+  # set to "recursive".
+  GIT_SUBMODULE_STRATEGY: "recursive"
 
 pages:
+  stage: deploy
   script:
-    # Install the zola package from the alpine community repositories
-    - apk add --update-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/community/ zola
-    # Execute zola build
+    - apk add zola
```
Comment (Keats) [inline docs/content/documentation/deployment/gitlab-pages.md]: Is it possible to specify a version?

Comment (walpox) [reply docs/content/documentation/deployment/gitlab-pages.md]: I have modified the file to allow setting a specific version of Zola.

Comment (walpox) [reply docs/content/documentation/deployment/gitlab-pages.md]: Note that Alpine only has the latest versions of packages (binary builds) due to storage costs and that the standalone `zola` executable from GitHub releases does not run in that distro container at all. I switched to the Debian image and downloading the executables from GitHub releases.

**Review by Keats — APPROVED** [review]

**Comment by Keats** [comment]
Thanks

**Pull request merged**